SeaPublicWebsite/ExternalServices/Models/Epb/EpbEpcDto.cs

```diff
@@ -58,4 +62,537 @@ public class EpbEpcAssessmentDto
+        {
+            return HomeAge.From1930To1949;
+        }
+        
+        if (PropertyAgeBand.Equals("D", StringComparison.OrdinalIgnoreCase) ||
+            PropertyAgeBand.Contains("1950-1966"))
+        {
+            return HomeAge.From1950To1966;
+        }
+        
+        if (PropertyAgeBand.Equals("E", StringComparison.OrdinalIgnoreCase) ||
+            PropertyAgeBand.Contains("1967-1975"))
+        {
+            return HomeAge.From1967To1975;
+        }
+        
+        if (PropertyAgeBand.Equals("F", StringComparison.OrdinalIgnoreCase) ||
+            PropertyAgeBand.Contains("1976-1982"))
+        {
+            return HomeAge.From1976To1982;
+        }
+        
+        if (PropertyAgeBand.Equals("G", StringComparison.OrdinalIgnoreCase) ||
+            PropertyAgeBand.Contains("1983-1990"))
+        {
+            return HomeAge.From1983To1990;
+        }
+        
+        if (PropertyAgeBand.Equals("H", StringComparison.OrdinalIgnoreCase) ||
+            PropertyAgeBand.Contains("1991-1995"))
+        {
+            return HomeAge.From1991To1995;
+        }
+        
+        if (PropertyAgeBand.Equals("I", StringComparison.OrdinalIgnoreCase) ||
+            PropertyAgeBand.Contains("1996-2002"))
+        {
+            return HomeAge.From1996To2002;
+        }
+        
+        if (PropertyAgeBand.Equals("J", StringComparison.OrdinalIgnoreCase) ||
+            PropertyAgeBand.Contains("2003-2006"))
+        {
+            return HomeAge.From2003To2006;
+        }
+        
+        if (PropertyAgeBand.Equals("K", StringComparison.OrdinalIgnoreCase) ||
+            PropertyAgeBand.Contains("2007-2011"))
+        {
+            return HomeAge.From2007To2011;
+        }
+        
+        if (PropertyAgeBand.Equals("L", StringComparison.OrdinalIgnoreCase) ||
+            PropertyAgeBand.Contains("2012 onwards"))
+        {
+            return HomeAge.From2012ToPresent;
+        }
+
+        return null;
+    }
+
+    private PropertyType? ParsePropertyType()
+    {
+        if (PropertyType is null)
+        {
+            return null;
+        }
+        
+        if (PropertyType.Contains("House", StringComparison.OrdinalIgnoreCase))
+        {
+            return BusinessLogic.Models.Enums.PropertyType.House;
+        }
+
+        if (PropertyType.Contains("Bungalow", StringComparison.OrdinalIgnoreCase))
+        {
+            return BusinessLogic.Models.Enums.PropertyType.Bungalow;
+        }
+
+        if (PropertyType.Contains("Flat", StringComparison.OrdinalIgnoreCase) ||
+            PropertyType.Contains("Maisonette", StringComparison.OrdinalIgnoreCase))
+        {
+            return BusinessLogic.Models.Enums.PropertyType.ApartmentFlatOrMaisonette;
+        }
+
+        return null;
+    }
+    
+    private HouseType? ParseHouseType()
+    {
+        if (ParsePropertyType() is not BusinessLogic.Models.Enums.PropertyType.House)
+        {
+            return null;
+        }
```

Comment:
For PropertyType and HasHotWaterCylinder, property names are the same as the type name and there's a conflicting namespace, so I have to use this entire type path. I had a thought of making these methods static and give them as argument the string / Dto object, which method do you think is bette?

Reply:
I think this is fine